idb/postgres/internal/writer/writer.go

```diff
@@ -125,24 +125,16 @@ func (w *Writer) Close() {
 	}
 }
 
-func (w *Writer) addBlockHeader(blockHeader *bookkeeping.BlockHeader) error {
-	_, err := w.tx.Exec(
-		context.Background(), addBlockHeaderStmtName,
+func addBlockHeader(blockHeader *bookkeeping.BlockHeader, batch *pgx.Batch) {
```

Comment:
1. I think it's good to preserve the distinction between add, write and update.
2. How would it work if these functions needed to be methods on `Writer` to e.g. access the log object?

Reply:
1. Up to you, it was just a thought when I noticed that it was queueing to the batch.
2. It wouldn't. In that case you would have to pass the log into the function instead of `*pgx.Batch`, or return an error so that you could log from the caller.